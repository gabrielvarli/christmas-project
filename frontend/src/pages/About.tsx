// const About = () => {
//   return (
//     <div className="flex flex-col justify-center items-center">
//       <div className="mt-20">
//         <h2 className="text-center text-5xl">
//           Din destination för julshopping 🎄
//         </h2>
//         <p className="text-center mt-4 sm:px-20 px-8">
//           På Christmas Shopping tror vi att julen är årets mest magiska tid – en
//           säsong fylld av glädje, värme och givmildhet. Vårt mål är att göra din
//           julshopping lika härlig som stressfri, med ett noggrant utvalt
//           sortiment av gåvor, dekorationer och julens nödvändigheter. <br></br>
//           Varför välja oss? 🎁 <br></br> Handplockade gåvor: Oavsett om du
//           handlar till nära och kära, vänner eller kollegor erbjuder vi ett
//           brett utbud av unika och meningsfulla presenter som passar alla smaker
//           och plånböcker. 🎄 Juldekorationer: Från glittrande ljus till eleganta
//           julgranskulor – vi har allt du behöver för att förvandla ditt hem till
//           ett vinterlandskap. 🛍️ Smidig shopping: Slipp julstressen! Vår
//           användarvänliga onlinebutik och snabba leveranser gör att du kan
//           fokusera på det som verkligen betyder något – att umgås med dina nära
//           och kära. 🌟 <br></br> Julstämning garanterad: Vi brinner för att
//           sprida julglädje och ser till att varje paket vi skickar är fyllt med
//           en liten touch av magi. <br></br>Vårt löfte Hos Christmas Shopping är
//           vi engagerade i kvalitet, service och julens anda. Varje produkt är
//           noggrant utvald, och vi tänker på varje detalj. Vi vill att din
//           julshopping ska vara lika glädjefylld som själva julafton. <br></br>
//           Tack för att du låter oss vara en del av dina jultraditioner.<br></br>{" "}
//           God Jul och ett Gott Nytt År! 🎅✨
//         </p>
//       </div>
//       <img src="/images/about.png" alt="about" className=" p-8 sm:p-20" />
//     </div>
//   );
// };

// export default About;

// // uppdatering

import React from "react";

const About: React.FC = () => {
  return (
    <section className="flex flex-col items-center justify-center px-4 sm:px-8 lg:px-16 py-12 bg-white text-gray-800">
      <div className="max-w-4xl text-center">
        <h2 className="text-3xl sm:text-5xl font-bold mb-6">
          Din destination för julshopping 🎄
        </h2>
        <p className="text-lg leading-relaxed">
          På{" "}
          <span className="font-semibold text-rose-600">
            Christmas Shopping
          </span>{" "}
          tror vi att julen är årets mest magiska tid – en säsong fylld av
          glädje, värme och givmildhet. Vårt mål är att göra din julshopping
          lika härlig som stressfri, med ett noggrant utvalt sortiment av gåvor,
          dekorationer och julens nödvändigheter.
        </p>
        <div className="mt-6">
          <h3 className="text-xl font-semibold text-rose-600 mb-4">
            Varför välja oss? 🎁
          </h3>
          <ul className="list-disc list-inside space-y-4 text-left mx-auto sm:mx-12">
            <li>
              <span className="font-semibold">Handplockade gåvor:</span> Oavsett
              om du handlar till nära och kära, vänner eller kollegor erbjuder
              vi ett brett utbud av unika och meningsfulla presenter som passar
              alla smaker och plånböcker.
            </li>
            <li>
              <span className="font-semibold">Juldekorationer:</span> Från
              glittrande ljus till eleganta julgranskulor – vi har allt du
              behöver för att förvandla ditt hem till ett vinterlandskap.
            </li>
            <li>
              <span className="font-semibold">Smidig shopping:</span> Slipp
              julstressen! Vår användarvänliga onlinebutik och snabba leveranser
              gör att du kan fokusera på det som verkligen betyder något – att
              umgås med dina nära och kära.
            </li>
            <li>
              <span className="font-semibold">Julstämning garanterad:</span> Vi
              brinner för att sprida julglädje och ser till att varje paket vi
              skickar är fyllt med en liten touch av magi.
            </li>
          </ul>
        </div>
        <p className="mt-6 text-lg leading-relaxed">
          Hos{" "}
          <span className="font-semibold text-rose-600">
            Christmas Shopping
          </span>{" "}
          är vi engagerade i kvalitet, service och julens anda. Varje produkt är
          noggrant utvald, och vi tänker på varje detalj. Vi vill att din
          julshopping ska vara lika glädjefylld som själva julafton.
        </p>
        <p className="mt-4 font-semibold text-rose-600">
          Tack för att du låter oss vara en del av dina jultraditioner. <br />
          God Jul och ett Gott Nytt År! 🎅✨
        </p>
      </div>
      <div className="mt-8">
        <img
          src="/images/about.png"
          alt="about"
          className="rounded-lg shadow-lg max-w-full h-auto"
        />
      </div>
    </section>
  );
};

export default About;
