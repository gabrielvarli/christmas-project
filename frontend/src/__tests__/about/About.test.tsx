//funkar

// import { render, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import About from "../../pages/About";

// describe("About page", () => {
//   it("should have a title stating 'Din destination f√∂r julshopping üéÑ'", () => {
//     render(<About />);
//     const heading: HTMLElement = screen.getByRole("heading", {
//       name: /Din destination f√∂r julshopping üéÑ/i,
//     });
//     expect(heading).toBeInTheDocument();
//   });

//   it("should have a picture of the about", () => {
//     render(<About />);
//     const imgElement: HTMLElement = screen.getByRole("img", {
//       name: /about/i,
//     });
//     expect(imgElement).toHaveAttribute("src", "/images/about.png");
//   });
// });

// // 1 error

// import { render, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import About from "../../pages/About";

// describe("About Page", () => {
//   let container: HTMLElement;

//   beforeEach(() => {
//     const renderResult = render(<About />);
//     container = renderResult.container; // Spara hela renderade outputen f√∂r snapshot
//   });

//   it("renders the title 'Din destination f√∂r julshopping üéÑ'", () => {
//     const heading = screen.getByRole("heading", {
//       name: /Din destination f√∂r julshopping üéÑ/i,
//     });
//     expect(heading).toBeInTheDocument();
//   });

//   it("displays the about image with the correct alt text and source", () => {
//     const imgElement = screen.getByRole("img", { name: /about image/i });
//     expect(imgElement).toBeInTheDocument();
//     expect(imgElement).toHaveAttribute("src", "/images/about.png");
//   });

//   it("matches the snapshot", () => {
//     expect(container).toMatchSnapshot(); // J√§mf√∂r det renderade resultatet med tidigare snapshot
//   });
// });

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import About from "../../pages/About";

describe("About Page", () => {
  it("displays the about image with the correct alt text and source", () => {
    render(<About />);

    const imgElement = screen.getByRole("img", { name: /about/i });
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", "/images/about.png");
  });

  it("displays the main heading correctly", () => {
    render(<About />);

    const headingElement = screen.getByRole("heading", {
      name: /Din destination f√∂r julshopping üéÑ/i,
    });
    expect(headingElement).toBeInTheDocument();
  });

  it("renders the reasons to choose Christmas Shopping", () => {
    render(<About />);

    const reasons = screen.getByText(/Varf√∂r v√§lja oss\? üéÅ/i);
    expect(reasons).toBeInTheDocument();
  });

  it("displays the closing message", () => {
    render(<About />);

    const closingMessage = screen.getByText(
      /Tack f√∂r att du l√•ter oss vara en del av dina jultraditioner\./i
    );
    expect(closingMessage).toBeInTheDocument();
  });
});
