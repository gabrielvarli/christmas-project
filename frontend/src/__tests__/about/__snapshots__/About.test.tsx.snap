// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page matches the snapshot 1`] = `
<div>
  <section
    class="flex flex-col items-center justify-center px-4 sm:px-8 lg:px-16 py-12 bg-white text-gray-800"
  >
    <div
      class="max-w-4xl text-center"
    >
      <h2
        class="text-3xl sm:text-5xl font-bold mb-6"
      >
        Din destination f√∂r julshopping üéÑ
      </h2>
      <p
        class="text-lg leading-relaxed"
      >
        P√•
         
        <span
          class="font-semibold text-rose-600"
        >
          Christmas Shopping
        </span>
         
        tror vi att julen √§r √•rets mest magiska tid ‚Äì en s√§song fylld av gl√§dje, v√§rme och givmildhet. V√•rt m√•l √§r att g√∂ra din julshopping lika h√§rlig som stressfri, med ett noggrant utvalt sortiment av g√•vor, dekorationer och julens n√∂dv√§ndigheter.
      </p>
      <div
        class="mt-6"
      >
        <h3
          class="text-xl font-semibold text-rose-600 mb-4"
        >
          Varf√∂r v√§lja oss? üéÅ
        </h3>
        <ul
          class="list-disc list-inside space-y-4 text-left mx-auto sm:mx-12"
        >
          <li>
            <span
              class="font-semibold"
            >
              Handplockade g√•vor:
            </span>
             Oavsett om du handlar till n√§ra och k√§ra, v√§nner eller kollegor erbjuder vi ett brett utbud av unika och meningsfulla presenter som passar alla smaker och pl√•nb√∂cker.
          </li>
          <li>
            <span
              class="font-semibold"
            >
              Juldekorationer:
            </span>
             Fr√•n glittrande ljus till eleganta julgranskulor ‚Äì vi har allt du beh√∂ver f√∂r att f√∂rvandla ditt hem till ett vinterlandskap.
          </li>
          <li>
            <span
              class="font-semibold"
            >
              Smidig shopping:
            </span>
             Slipp julstressen! V√•r anv√§ndarv√§nliga onlinebutik och snabba leveranser g√∂r att du kan fokusera p√• det som verkligen betyder n√•got ‚Äì att umg√•s med dina n√§ra och k√§ra.
          </li>
          <li>
            <span
              class="font-semibold"
            >
              Julst√§mning garanterad:
            </span>
             Vi brinner f√∂r att sprida julgl√§dje och ser till att varje paket vi skickar √§r fyllt med en liten touch av magi.
          </li>
        </ul>
      </div>
      <p
        class="mt-6 text-lg leading-relaxed"
      >
        Hos
         
        <span
          class="font-semibold text-rose-600"
        >
          Christmas Shopping
        </span>
         
        √§r vi engagerade i kvalitet, service och julens anda. Varje produkt √§r noggrant utvald, och vi t√§nker p√• varje detalj. Vi vill att din julshopping ska vara lika gl√§djefylld som sj√§lva julafton.
      </p>
      <p
        class="mt-4 font-semibold text-rose-600"
      >
        Tack f√∂r att du l√•ter oss vara en del av dina jultraditioner. 
        <br />
        God Jul och ett Gott Nytt √Ör! üéÖ‚ú®
      </p>
    </div>
    <div
      class="mt-8"
    >
      <img
        alt="about"
        class="rounded-lg shadow-lg max-w-full h-auto"
        src="/images/about.png"
      />
    </div>
  </section>
</div>
`;
